docs/reference-implementation/Api-Management.md

```diff
@@ -0,0 +1,75 @@
+# Adding API Management
+
+A common scenario, which is not covered by the main reference implementation, is to include an API Management service in the architecture. It is not included in the main branch as it is considered an optional service and not essentially required in many cases.
+
+Therefore there is a [separate branch](https://github.com/Azure/Mission-Critical-Online/tree/feature/api-management) which includes Azure API Management (APIM) and all the required changes to properly embed this in the reference implementation. This article walks through the various required changes.
+
+## Architecture overview
+
+The architecture diagram for Azure Mission-Critical Online using an regional APIM instance is shown below.
+
+![Mission-Critical architecture diagram with API Management](/docs/media/Mission-critical-APIM-architecture.svg)
+
+It is important to note that API Management is used as a regional resource, i.e. deployed per stamp, and not in the geo-replicated version of APIM. This is to not break the principle of having no regional dependencies wherever possible. While the geo-replicated deployment of APIM does provide resiliency of the data plane (i.e. the gateways) against a regional outage, the control plane is bound to the primary region. If APIM in that region experiences issues, no deployments/changes can be made. Hence, we deploy separate instances per stamp.
```

Comment:
Maybe reiterate here that this approach requires the API definitions etc. to be managed with IaC too, otherwise you'd have to make changes in all instances?

What would be the use case for geo-replicated APIM? "Less mission-critical" deployments which can deal with the inability to modify the APIs, while the system still works in an outage...

Reply:
the only benefit of a geo replicated APIM I can see if people might be working more handson with the developer portal?! maybe then it is better to only have one instance? but not even sure about that since you still should have AFD in front